packages/flutter_test/lib/src/recording_canvas.dart

```diff
@@ -170,7 +173,17 @@ class TestRecordingPaintingContext extends ClipContext implements PaintingContex
     canvas.saveLayer(null, Paint()); // TODO(ianh): Expose the alpha somewhere.
     painter(this, offset);
     canvas.restore();
-    return OpacityLayer();
+    final OpacityLayer layer = OpacityLayer();
+    _createdLayers.add(layer);
+    return layer;
+  }
+
+  /// Releases allocated resources.
+  @mustCallSuper
+  void dispose() {
+    for (final OpacityLayer layer in _createdLayers) {
+      layer.dispose();
+    }
```

Comment:
This doesn't look right. I believe, the caller of pushOpacity assumes ownership of the OpacityLayer and should be in charge of disposing it, not the PaintContext.

Reply:
This thing is just for testing. 
And, i think it is reasonable to ask test to dispose the instance of created TestRecordingPaintingContext, at the end of the test.

Otherwise, it should return the layer somehow, that means change of contract.